main/src/mpu.cpp

```diff
@@ -32,60 +27,40 @@ void init() {
     // mpudev.setZGyroOffset(0);
 
     if (!mpudev.testConnection()) {
-        // die
         Serial.println("Connection to mpu failed");
         return;
     }
 
-    if (mutex == nullptr) {
-        // die
-        Serial.println("Couldn't allocate resources for mutex!");
-        return;
-    }
-
     init_success = true;
     accel_res = mpudev.get_acce_resolution() / 2;
     gyro_res = mpudev.get_gyro_resolution() / 2;
 }
 
-/**
- * @brief Consume and return collected data.
- *        ! CAN BLOCK INDEFINITELY (if INCLUDE_vTaskSuspend == 1) !
- * @return Max g force, Average g force, Average rotation.
- */
-Data get_mpu() {
-    xSemaphoreTake(mutex, portMAX_DELAY);
-
-    Data dane{
-        .max_x = internal.max_x,
-        .max_y = internal.max_y,
-        .max_z = internal.max_z,
-        .avg_x = internal.avg_x / count, // Instead of counting could implement
-        .avg_y = internal.avg_y / count, // moving average, but dunno if it's
-        .avg_z = internal.avg_z / count, // superior and/or viable alternative.
-        .rot_x = internal.rot_x / count,
-        .rot_y = internal.rot_y / count,
-        .rot_z = internal.rot_z / count};
+Data get_data() {
+    Data dane{};
+    dane.max.x = internal.max.x;
+    dane.max.y = internal.max.y;
+    dane.max.z = internal.max.z;
+    dane.avg.x = internal.avg.x / count,
+    dane.avg.y = internal.avg.y / count,
+    dane.avg.z = internal.avg.z / count,
+    dane.rot.x = internal.rot.x / count,
+    dane.rot.y = internal.rot.y / count,
+    dane.rot.z = internal.rot.z / count,
 
     count = 0;
     internal = {};
 
-    xSemaphoreGive(mutex);
-
     return dane;
 }
 
-/**
- * @brief Main MPU task that collects data from the sensor.
- *        Die gracefully if sensor not initialised.
- *        ! CAN BLOCK INDEFINITELY (if INCLUDE_vTaskSuspend == 1) !
- */
 void mpu_task([[maybe_unused]] void *pvParameters) {
     int16_t ax, ay, az, gx, gy, gz;
     float fax, fay, faz, fgx, fgy, fgz;
 
-    if (!init_success)
-        vTaskDelete(NULL);
+    if (!init_success) {
+        vTaskDelete(nullptr);
+    }

```

Comment:
Now it fails silently, but it should *somehow* inform us that it failed

Reply:
Task fails if init failed. Init is verbose about failure.

Reply:
Deleting a whole task silently is a bad idea even if it won't work properly

Reply:
So it should print that it failed and then be removed?

Reply:
For now yes, we are planning to add some form of error handling and recovery capability but that's outside of scope for this task